newrelic/resource_newrelic_cloud_gcp_integrations.go

```diff
@@ -0,0 +1,1642 @@
+	} else if o, n := d.GetChange("router"); len(n.([]interface{})) < len(o.([]interface{})) {
+		gcpDisableIntegrations.GcpRouter = []cloud.CloudDisableAccountIntegrationInput{{LinkedAccountId: linkedAccountID}}
+	}
+
+	if v, ok := d.GetOk("run"); ok {
+		gcpCloudIntegrations.GcpRun = expandCloudGcpRunIntegrationsInputs(v.([]interface{}), linkedAccountID)
+	} else if o, n := d.GetChange("run"); len(n.([]interface{})) < len(o.([]interface{})) {
+		gcpDisableIntegrations.GcpRun = []cloud.CloudDisableAccountIntegrationInput{{LinkedAccountId: linkedAccountID}}
+	}
+
+	if v, ok := d.GetOk("spanner"); ok {
+		gcpCloudIntegrations.GcpSpanner = expandCloudGcpSpannerIntegrationsInputs(v.([]interface{}), linkedAccountID)
+	} else if o, n := d.GetChange("spanner"); len(n.([]interface{})) < len(o.([]interface{})) {
+		gcpDisableIntegrations.GcpSpanner = []cloud.CloudDisableAccountIntegrationInput{{LinkedAccountId: linkedAccountID}}
+	}
+
+	if v, ok := d.GetOk("sql"); ok {
+		gcpCloudIntegrations.GcpSql = expandCloudGcpSQLIntegrationsInputs(v.([]interface{}), linkedAccountID)
+	} else if o, n := d.GetChange("sql"); len(n.([]interface{})) < len(o.([]interface{})) {
+		gcpDisableIntegrations.GcpSql = []cloud.CloudDisableAccountIntegrationInput{{LinkedAccountId: linkedAccountID}}
+	}
+
+	if v, ok := d.GetOk("storage"); ok {
+		gcpCloudIntegrations.GcpStorage = expandCloudGcpStorageIntegrationsInputs(v.([]interface{}), linkedAccountID)
+	} else if o, n := d.GetChange("storage"); len(n.([]interface{})) < len(o.([]interface{})) {
+		gcpDisableIntegrations.GcpStorage = []cloud.CloudDisableAccountIntegrationInput{{LinkedAccountId: linkedAccountID}}
+	}
+
+	if v, ok := d.GetOk("virtual_machines"); ok {
+		gcpCloudIntegrations.GcpVms = expandCloudGcpVirtualMachinesIntegrationsInputs(v.([]interface{}), linkedAccountID)
+	} else if o, n := d.GetChange("virtual_machines"); len(n.([]interface{})) < len(o.([]interface{})) {
+		gcpDisableIntegrations.GcpVms = []cloud.CloudDisableAccountIntegrationInput{{LinkedAccountId: linkedAccountID}}
+	}
+
+	if v, ok := d.GetOk("vpc_access"); ok {
+		gcpCloudIntegrations.GcpVpcaccess = expandCloudGcpVpcAccessIntegrationsInputs(v.([]interface{}), linkedAccountID)
+	} else if o, n := d.GetChange("vpc_access"); len(n.([]interface{})) < len(o.([]interface{})) {
+		gcpDisableIntegrations.GcpVpcaccess = []cloud.CloudDisableAccountIntegrationInput{{LinkedAccountId: linkedAccountID}}
+	}
+
+	configureInput := cloud.CloudIntegrationsInput{
+		Gcp: gcpCloudIntegrations,
+	}
+	disableInput := cloud.CloudDisableIntegrationsInput{
+		Gcp: gcpDisableIntegrations,
+	}
+	return configureInput, disableInput
+}
+
+//expand function to extract inputs from gcp app engine schema
+func expandCloudGcpAppEngineIntegrationsInputs(b []interface{}, linkedAccountID int) []cloud.CloudGcpAppengineIntegrationInput {
+	expanded := make([]cloud.CloudGcpAppengineIntegrationInput, len(b))
+	for i, expand := range b {
+		var Input cloud.CloudGcpAppengineIntegrationInput
```

Comment:
```suggestion
		var input cloud.CloudGcpAppengineIntegrationInput
```

Reply:
Per Go convention, variables should be lower cases unless we want to export them to another package